tests/frontend/test_program.py

```diff
@@ -256,8 +263,14 @@ def test_assert_max_number_of_measurements(self, measure_op, measure_name):
     def test_assert_max_number_of_measurements_wrong_entry(self):
         """Check that the correct error is raised when calling `prog.assert_number_of_measurements`
         with the incorrect type of device spec mode entry."""
-        device_dict = {"modes": 2, "layout": None, "gate_parameters": None, "compiler": [None]}
-        spec = sf.api.DeviceSpec(target="simulon_gaussian", connection=None, spec=device_dict)
+        device_dict = {
+            "target": "simulon_gaussian",
+            "modes": 2,
+            "layout": None,
+            "gate_parameters": None,
+            "compiler": [None],
```

Comment:
Are you sure this is legal?
```suggestion
            "compiler": [],
```

Reply:
Hmm, you're right. Not really, but it made the tests pass (since prior the compiler was set to `None` in the tests).

Reply:
Fixed so that `"layout"` is always a string, `"gate_parameters"` is always a dictionary, and `"compiler"` is always a list of strings or empty list.